File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDataViewModel.java
package com.cajetan.youtubeplus.data;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import java.util.List;

public class VideoDataViewModel extends AndroidViewModel {
    private VideoDataRepository mRepository;
    private LiveData<List<VideoData>> mAllVideoData;

    public VideoDataViewModel(Application application) {
        super(application);
        mRepository = new VideoDataRepository(application);
        mAllVideoData = mRepository.getAllVideoData();
    }

    public LiveData<List<VideoData>> getAllVideoData() {
        return mAllVideoData;
    }

    public void insert(VideoData videoData) {
        mRepository.insert(videoData);
    }

    public void delete(VideoData videoData) {
        mRepository.delete(videoData);
    }
}

File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDataViewModel.kt
package com.cajetan.youtubeplus.data

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class VideoDataViewModel(val context: Application) : AndroidViewModel(context) {
    private var mRepository: VideoDataRepository = VideoDataRepository(context)
    private var mAllVideoData: LiveData<List<VideoData>>

    init {
        mAllVideoData = mRepository.getAllVideoData()
    }

    fun getAllVideoData(): LiveData<List<VideoData>> {
        return mAllVideoData
    }

    fun contains(id: String): Boolean {
        return mRepository.contains(id)
    }

    fun insert(videoData: VideoData) {
        mRepository.insert(videoData)
    }

    fun delete(videoData: VideoData) {
        mRepository.delete(videoData)
    }
}